### ISSUE TITLE
Incoming Call VOIP notification not fired when app is in background or killed state in Production build

### ISSUE THREAD

**Issue by BCSingh** [QuickBlox/quickblox-ios-sdk#1223]
**New Issue Checklist**

 - [YES] I have read the [Documentation](https://quickblox.com/developers/IOS)
 - [YES ] I have searched for a similar issue in the [project](https://github.com/QuickBlox/quickblox-ios-sdk/issues) and found none

**Environment details**

Info                           | Value                                       |
-------------------------------|---------------------------------------------|
iOS Version                    | 13.4.1                                    |                                          
Quickblox iOS SDK version      |  2.17.4                                  |
QuickbloxWebRTC SDK version    | 2.7.5                                 |                               
Xcode Version                  | Xcode 11.5                              |
Repro with our demo sample     | it has issues to upload to testflight   |

**Did this work before?**
No, it never worked in production build. but it works in development mode when i connect my device and run the app directly on it

**Expected behavior**
Incoming VOIP call notifications should work when the app is in all foreground, background and killed states. 

**Actual behavior**
In development build while device is connected and being debugged, incoming call is received in all these states.
In Production build uploaded to TestFlight, it works only in Foreground. Background and killed states never get incoming call notification.

**Logs**

Logs not available in Production build. But in Quickblox admin panel, the notification queue says that the notification is delivered.

**Steps to reproduce the behavior**
- Upload the app to Testflight.
- Make a voip call to the app.
- App doesn't receive voip notification

**Any others comments**
I kept changing the certificates in Quickblox admin portal and tested.  It never worked though.  I suspect if it's an iOS restriction to not allow incoming VOIP call notification?  But how does it work in development mode?
When an incoming call comes to the app, backend sends this Quickblox VOIP notification and a firebase apns notification.  I receive the firebase apns notification alone everytime. and in development mode, i receive the apns first and then the voip call. 
 I don't think firing both notification would be an issue. 
I have copied exact code snippets from sample app. So I need to know will the sample work on production with incoming call? has it been tested. for it? I get many issues in the sample app, so i couldnt upload it to testflight.  And any quick help on this issue? We are waiting to push it to Production.

in Quickblox admin panel, i tried this 3 different combination of certificates:
1. Apple Development iOS Push **and** VOIP 
2. VOIP
3. Apple Push Services

**Comment by kirti301290** [comment]
@BCSingh @QB-vladyslav-vyshniak 
can help facing same problem  in production and working fine in development mode
Thanks

**Comment by BCSingh** [comment]
Ok I got it solved. Guessing it to be either a logical issue or the certificate issue. There are few things I did to get this solved. see if any of these points helps

- In quickblox admin portal, i created 2 apps. I used one for development and uploaded 2 certificates -  Apple Development iOS Push and VOIP p12 files. I used the second app production by uploading just the Apple Push services(Production) p12 . I configured the app to use the first app credentials in DEV environment and the second app for Production environment.
- When app opens, do Quickblox initialization with the set of codes given in the doc. `QBRTCClient.initializeRTC()`
- Save the generated APNS token and VOIP token locally in their corresponding delegates - `didRegisterForRemoteNotificationsWithDeviceToken` and `pushRegistry(_ registry: PKPushRegistry, didUpdate`
- When the user signup to the app, the backend signup API must be creating a quickblox user and returning the `quickblox user ID` to the app. Save this locally. Next time when we login to the app, we should receive this ID and save it locally.
- Now after login/signups success or everytime you open the app, check if we have the quickblox ID and do these 3 things:
      -  Do quickblox login `QBRequest.logIn`. 
      -  On login success, connect to chat to keep it open for incoming VOIP pushes in foreground. 
      -  Create QB subscriptions for APNS and VOIP.
If these 3 steps are success, then we get incoming calls in all states and all environments.

These steps worked for me. Some similar logic should work for all.
